Provider/MenuProvider.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Grossum\MenuBundle\Provider;
+
+use Knp\Menu\FactoryInterface;
+use Knp\Menu\ItemInterface;
+use Knp\Menu\Provider\MenuProviderInterface;
+
+use Grossum\MenuBundle\Entity\BaseMenu;
+use Grossum\MenuBundle\Entity\EntityManager\BaseMenuManager;
+use Grossum\MenuBundle\Entity\EntityManager\BaseMenuItemManager;
+use Grossum\MenuBundle\Manager\MenuManager;
+
+class MenuProvider implements MenuProviderInterface
+{
+    /**
+     * @var FactoryInterface
+     */
+    protected $factory;
+
+    /**
+     * @var BaseMenuManager
+     */
+    protected $menuManager;
+
+    /**
+     * @var BaseMenuItemManager
+     */
+    protected $menuItemManager;
+
+    /**
+     * @var MenuManager
+     */
+    private $menuMaster;
+
+    /**
+     * @param FactoryInterface $factory
+     * @param BaseMenuManager $menuManager
+     * @param BaseMenuItemManager $menuItemManager
+     * @param MenuManager $menuMaster
+     */
+    public function __construct(
+        FactoryInterface $factory,
+        BaseMenuManager $menuManager,
+        BaseMenuItemManager $menuItemManager,
+        MenuManager $menuMaster
+    ) {
+        $this->factory         = $factory;
+        $this->menuManager     = $menuManager;
+        $this->menuItemManager = $menuItemManager;
+        $this->menuMaster      = $menuMaster;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get($name, array $options = [])
+    {
+        $menuEntity = $this->find($name);
+
+        if ($menuEntity === null) {
+            throw new \InvalidArgumentException(sprintf('The menu "%s" is not defined.', $name));
+        }
+
+        $repo = $this->menuItemManager->getRepository();
+
+        $rootMenuEntity = $repo->findMenuRootItem($menuEntity);
+        $treeNodes      = $repo->getNodesHierarchy($rootMenuEntity);
+        $nestedTree     = $repo->buildTreeArray($treeNodes);
+        $childrenIndex  = $repo->getChildrenIndex();
+
+        $build = function (array $tree, ItemInterface $parent) use (&$build, $childrenIndex) {
```

Comment:
It's a very strange usage.
You create `Closure` and pass this closure in self by using `use`.

Why do you not place in as a separate method?